service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCMOControllerTest.java

```diff
@@ -0,0 +1,163 @@
+    private static final String CASE_MANAGEMENT_ORDER_KEY = "caseManagementOrder";
+
+    private static final byte[] pdf = {1, 2, 3, 4, 5};
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    private Document document;
+
+    @BeforeEach
+    void setup() throws IOException {
+        document = document();
+        DocmosisDocument docmosisDocument = new DocmosisDocument("case-management-order.pdf", pdf);
+
+        given(docmosisDocumentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
+        given(uploadDocumentService.uploadPDF(any(),any(), any(), any())).willReturn(document);
+    }
+
+    @Test
+    void aboutToStartShouldReturnDraftCaseManagementOrderForAction() throws Exception {
+        CallbackRequest request = buildCallbackRequest(createDraftCaseManagementOrder());
+
+        MvcResult response = makeRequest(request, "about-to-start");
+        AboutToStartOrSubmitCallbackResponse callbackResponse = objectMapper.readValue(
+            response.getResponse().getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData convertedCasData = objectMapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        CaseManagementOrder caseManagementOrder = convertedCasData.getCaseManagementOrder();
+        assertThat(caseManagementOrder).isEqualTo(createDraftCaseManagementOrder());
+    }
+
+    @Test
+    void midEventShouldReturnReviewedDocumentReferenceForAction() throws Exception {
+        CallbackRequest request = buildCallbackRequest(createDraftCaseManagementOrder());
+
+        MvcResult response = makeRequest(request, "mid-event");
+        AboutToStartOrSubmitCallbackResponse callbackResponse = objectMapper.readValue(
+            response.getResponse().getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        verify(uploadDocumentService).uploadPDF(USER_ID, AUTH_TOKEN, pdf, "draft-case-management-order.pdf");
+
+        Map<String, Object> responseCaseData = callbackResponse.getData();
+
+        assertThat(responseCaseData).containsKey(CASE_MANAGEMENT_ORDER_ACTION_KEY);
+
+        Map<String, Object> caseManagementOrderAction = (Map<String, Object>) responseCaseData.get(
+            CASE_MANAGEMENT_ORDER_ACTION_KEY);
+        assertThat(caseManagementOrderAction).containsEntry(
+            "orderDoc", ImmutableMap.builder()
+                .put("document_binary_url", document.links.binary.href)
+                .put("document_filename", document.originalDocumentName)
+                .put("document_url", document.links.self.href)
+                .build());
+    }
+
+    @Test
+    void aboutToSubmitShouldReturnCaseDataWithApprovedCaseManagementOrder() throws Exception {
```

Comment:
should it return approved case management order?

Reply:
what if it isn't approved? I guess returns judge's actions would be better summary?